### DISCUSSION TITLE
[Implemented] /cmi tree through attachcommand !cc!

### DISCUSSION THREAD

**Discussion by mrfloris** [Zrips/CMI#1160] [Feature Requests]
Spigot 1.13.2 and CMI 8.2.x

Could we perhaps get the ability to use !cc!cmi tree jungle_bush through attachcommand? 

We might want to let players buy custom tools with limited use of this, to help make building bigger areas more fun and quicker. But you have to run it as the player. We however cannot really give the permission, because that means they can just use the /cmi tree command and bypass the limited use, also, the https://www.spigotmc.org/wiki/cmi-permissions/ doesn't seem to have the individual permissions listed, i dont even know if it's possible. 

Anyway.. it would be nice if it could recognize the block they're interacting with.

**Comment by sandwally** [comment]
Can almost be done
/cmi attachcommand !limiteduse:5!;;!silent!!cc!lp user [playerName] permission settemp true 2s cmi.command.tree;;!silent!cmi tree jungle_bush
But the command cmi tree jungle_brush will be done before temp permission and I don't know if you can add a delay on attachcommand

So a way to make console run commands on/as player is needed.
Or a delay that we can use on attachcommands or make the commands run in sequence of the order on the ac command.

**Comment by mrfloris** [comment]
I tried to make one for hat today, to let players wear an event item as a hat, same thing 'must be run as player'. it would be nice if these could be run as console ;)

**Comment by LogGits** [comment]
Maybe a new attribute `!gp!` meaning GrantPermission allowing the player to use the command by using the item. This would be useful as it would allow running other commands as well 👍

**Comment by mrfloris** [comment]
**This has been implemented,** 
`asConsole! cmi tree Acacia -p:[playerName]`
This works for me on the new way that `/cmi attachcommand` works.
<img width="1975" alt="Screen-Shot 2022-09-15 at 17 14 54" src="https://user-images.githubusercontent.com/28841349/190441777-6c96c273-bf27-46fe-9aa4-3d70de9bd819.png">
